impl/NotSoBasicLinearAlgebra.h

```diff
@@ -322,6 +323,46 @@ typename ParentType::DType Determinant(const MatrixBase<ParentType, Dim, Dim, ty
     return det;
 }
 
+template <typename ParentType, int Dim>
+typename std::enable_if<std::is_integral<typename ParentType::DType>::value, typename ParentType::DType>::type
+Determinant(const MatrixBase<ParentType, Dim, Dim, typename ParentType::DType> &A)
+{
+    // For integral types use Bareiss algorithm
+    Matrix<Dim, Dim, typename ParentType::DType> A_copy = A;
+
+    int sign = 1;
+    typename ParentType::DType prev = 1;
+
+    for (int i = 0; i < Dim; i++)
+    {
+        if (A_copy(i, i) == 0)
+        {
+            int idx = i + 1;
+            for (; idx < Dim; idx++)
+            {
+                if (A_copy(idx, i) != 0) break;
+            }
+            if (idx == Dim) return 0;
+            const BLA::Matrix<1, Dim, typename ParentType::DType> tmp = A_copy.Row(i);
```

Comment:
```suggestion
            const auto tmp = A_copy.Row(i);
```
Row will return a `RefMatrix` which doesn't copy the original array. Since you're not modifying tmp then it's probably a bit more efficient to avoid taking a copy here

Reply:
Sorry, I'm a bit confused, and the code is too. We are trying to swap the two rows, I think for that we need a temporary copy. I cannot use the `std::swap` or `bla_swap` methods because the assignment operator is deleted for `RefMatrix`. Do I have something wrong?

Also, I'm not sure if the `auto` will make a `RefMatrix` or `Matrix` type.

Reply:
Ahh my mistake! Sorry, I should have read this more closely. No problems, it makes sense to take a copy if you're swapping rows :+1: